File: 剑指offer/Python/1.py
# -*- coding:utf-8 -*-
class Solution:
    # array 二维列表
    def Find(self, target, array):
        # write code here
        row = len(array)
        col = len(array[0])
        i = 0
        j = col - 1
        while i < row and j >= 0:
            if array[i][j] < target:
                i += 1
            elif array[i][j] > target:
                j -= 1
            else:
                return True
        return False

File: point-to-offer/CPP/1.cpp
#include <vector>

class Solution {
public:
    bool Find(int target, vector<vector<int> > array) 
    {
        int row = array.size();
        int col = array.front().size();

        int i = 0;
        int j = col - 1;
        while (i < row && j >= 0)
        {
            if (array[i][j] < target)
            {
                i++;
            }
            else if (array[i][j] > target)
            {
                j--;
            }
            else 
            {
                return true;
            }
        }

        return false;
    }
};